src/deepsparse/loggers/metric_functions/built_ins.py

```diff
@@ -11,12 +11,192 @@
+        Assumptions:
+            - 3 dimensional or 4 dimensional (num_batches in zeroth dimension)
+              tensor/array
+            - the image has 3 or 1 channels
+    :return: Tuple containing the mean pixel values:
+        - 3 floats if image has 3 channels
+        - 1 float if image has 1 channel
+    """
+    img_numpy = _assert_numpy_image(img)
+    num_dims, channel_dim = _check_valid_image(img_numpy)
+    dims = numpy.arange(0, num_dims, 1)
+    dims = numpy.delete(dims, channel_dim)
+    return tuple(numpy.mean(img_numpy, axis=tuple(dims)))
+
+
+def std_pixels_per_channel(
+    img: Union[numpy.ndarray, torch.tensor]
+) -> Union[Tuple[float, float, float], Tuple[float]]:
+    """
+    Return the standard deviation of pixel values per image channel
+    :param img: An image represented as a numpy array or a torch tensor.
+        Assumptions:
+            - 3 dimensional or 4 dimensional (num_batches in zeroth dimension)
+              tensor/array
+            - the image has 3 or 1 channels
+    :return: Tuple containing the standard deviation of pixel values:
+        - 3 floats if image has 3 channels
+        - 1 float if image has 1 channel
+    """
+    img_numpy = _assert_numpy_image(img)
+    num_dims, channel_dim = _check_valid_image(img)
+    dims = numpy.arange(0, num_dims, 1)
+    dims = numpy.delete(dims, channel_dim)
+    return tuple(numpy.std(img_numpy, axis=tuple(dims)))
+
+
+def max_pixels_per_channel(
+    img: Union[numpy.ndarray, torch.tensor]
+) -> Union[Tuple[float, float, float], Tuple[float]]:
+    """
+    Return the max pixel value per image channel
+    :param img: An image represented as a numpy array or a torch tensor.
+        Assumptions:
+            - 3 dimensional or 4 dimensional (num_batches in zeroth dimension)
+              tensor/array
+            - the image has 3 or 1 channels
+    :return: Tuple containing the max pixel values:
+        - 3 floats if image has 3 channels
+        - 1 float if image has 1 channel
+    """
+    img_numpy = _assert_numpy_image(img)
+    num_dims, channel_dim = _check_valid_image(img)
+    dims = numpy.arange(0, num_dims, 1)
+    dims = numpy.delete(dims, channel_dim)
+    return tuple(numpy.max(img_numpy, axis=tuple(dims)))
+
+
+def fraction_zeros(img: Union[numpy.ndarray, torch.tensor]) -> float:
+    """
+    Return the float the represents the fraction of zeros in the
+    image tensor/array
+
+    :param img: An image represented as a numpy array or a torch tensor.
+       Assumptions:
+           - 3 dimensional or 4 dimensional (num_batches in zeroth dimension)
+             tensor/array
+           - the image has 3 or 1 channels
+    :return: A float in range from 0. to 1.
+    """
+    image_numpy = _assert_numpy_image(img)
+    _check_valid_image(image_numpy)
+    return (image_numpy.size - numpy.count_nonzero(image_numpy)) / image_numpy.size
+
+
+def bounding_box_count(bboxes: List[List[Optional[List[float]]]]) -> Dict[int, int]:
```

Comment:
We should probably support the yolact and yolo outputs directly here. Additionally a generic len or shape function would be useful

Reply:
Yeah, those are in the end the yolact and yolo outputs (`List[List[Optional[List[float]]]]` for yolo and `List[List[List[float]]]` for yolact). Let's land now, we will be revisiting the built-ins anyway.